onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorConvolution.cpp

```diff
@@ -73,7 +77,7 @@ class DmlOperatorConvolution : public DmlOperator, public ConvolutionHelperBase
         DML_CONVOLUTION_OPERATOR_DESC convDesc = {};
         convDesc.InputTensor = &inputDescs[0];
         convDesc.FilterTensor = &inputDescs[1];
-        convDesc.BiasTensor = kernelInfo.GetInputCount() > 2 ? &inputDescs[2] : nullptr;
+        convDesc.BiasTensor = isWithDynamicPads ? (kernelInfo.GetInputCount() > 3 ? &inputDescs[3] : nullptr) : (kernelInfo.GetInputCount() > 2 ? &inputDescs[2] : nullptr);
```

Comment:
Ugh, to whoever invented this silly operator, why didn't they just append the Pads tensor to the end of the existing inputs instead of inserting it into the middle? Since this model isn't live yet (AFAIK), can we instead fix the schema and model to something more sane? That would be the ideal thing to do.

Also, didn't we have a helper function (maybe it was deleted) that essentially did this, checking the index against the actual bounds and returning nullptr if it didn't exist? e.g. `TryGetTensorDesc(inputDescs, biasIndex);`

Reply:
Agreed, though it's out there and being used.  I think the ship has sailed.